### PR TITLE
Add specification for cl_ext_image_requirements_info

### PR DESCRIPTION
**Pull request by kpet**
Signed-off-by: Kevin Petit <kevin.petit@arm.com>
Change-Id: I42ed70f825d63cace455f48101d8b2ee1ed78c42

### CODE REVIEW

File: extensions/cl_ext_image_requirements_info.asciidoc [hunk 1]
```diff
@@ -0,0 +1,389 @@
...
+  * _flags_ is a bit-field that is used to specify allocation and usage information
+    about the image format being queried and is described in the
+    <<memory-flags-table,Memory Flags>> table. _flags_ may be {CL_MEM_READ_WRITE} to
+    specialize the query for images that may be read from and written to by different
+    kernel instances when correctly ordered by event dependencies, or {CL_MEM_READ_ONLY}
+    to specialize the query for images that may be read from by a kernel, or
+    {CL_MEM_WRITE_ONLY} to specialiaze the query for images that may be written to by
+    a kernel, or {CL_MEM_KERNEL_READ_AND_WRITE} to specialize the query for images that
+    may be both read from and written to by the same kernel instance. When _flags_ is
+    `0` the value returned for the query must be correct for a possible values of _flags_.
```
Comment (Jeremy-Kemp) [inline extensions/cl_ext_image_requirements_info.asciidoc:162]:
```suggestion
    `0` the value returned for the query must be correct for all possible values of _flags_.
```

Comment (kpet) [reply extensions/cl_ext_image_requirements_info.asciidoc:162]: Yup, thanks!

**Comment by bashbaug** [comment]
Could we merge this PR, or at least the XML changes?  I was updating my header generation branch and it took me a moment to figure out why the headers didn't contain this extension.  Thanks!

**Comment by kpet** [comment]
@bashbaug I've just resolved the conflicts in the XML. Happy for you to approve/merge this PR.

File: extensions/cl_ext_image_requirements_info.asciidoc [hunk 2]
```diff
@@ -0,0 +1,389 @@
...
+CL_IMAGE_REQUIREMENTS_BASE_ADDRESS_ALIGNMENT_EXT 0x1292
+CL_IMAGE_REQUIREMENTS_SIZE_EXT                   0x12B2
+CL_IMAGE_REQUIREMENTS_MAX_WIDTH_EXT              0x12B3
+CL_IMAGE_REQUIREMENTS_MAX_HEIGHT_EXT             0x12B4
+CL_IMAGE_REQUIREMENTS_MAX_DEPTH_EXT              0x12B5
+CL_IMAGE_REQUIREMENTS_MAX_ARRAY_SIZE_EXT         0x12B6
+----
+
+== Modifications to the OpenCL API Specification
+Modify Section 5.3.1, *Creating Image Objects*) ::
```
Comment (bashbaug) [inline extensions/cl_ext_image_requirements_info.asciidoc:94]: Editorial: the indentation is a little weird here, and there's a mis-matched parenthesis.

**Review by bashbaug — APPROVED** [review]
Merging as discussed in this thread.

I found one editorial issue that would be nice to fix before publishing this extension on the registry, but the specification is understandable even without it fixed.

There are also a few TODOs remaining in the document, that would be nice to resolve.

Thanks!

**Pull request merged**